packages/pool/contracts/DelegationUtils.sol

```diff
@@ -42,22 +42,22 @@ contract DelegationUtils is RewardUtils, IDelegationUtils {
         if (userDelegate != address(0)) {
             // Need to revoke previous delegation
             User storage prevDelegate = users[userDelegate];
-            prevDelegate.delegatedTo.push(Checkpoint({
-                fromBlock: block.number,
-                value: getValue(prevDelegate.delegatedTo) - userShares
-                }));
+            updateCheckpointArray(
+                prevDelegate.delegatedTo,
+                getValue(prevDelegate.delegatedTo) - userShares
+                );
         }
         // Assign the new delegation
         User storage _delegate = users[delegate];
-        _delegate.delegatedTo.push(Checkpoint({
-            fromBlock: block.number,
-            value: getValue(_delegate.delegatedTo) + userShares
-            }));
+        updateCheckpointArray(
+            _delegate.delegatedTo,
```

Comment:
This way we can potentially cast more tokens  after the vote was started .and add them to the voting
If we have this sequence of actions:
User Voted -> User staked more coins -> User delegated to another user

then the delegate will be able to vote on the vote with more coins thenuser had at the beginning of the vote....

If userShares aren't updated then we are fine though.

Reply:
I'm not sure if I'm following, can you write out the exact steps of the scenario? If they have delegated to another user after the proposal is made, then their delegation won't count for that proposal.

Reply:
Oh, that was exactly what I wanted to understand, I think because we are getting checkpoint with the old blocknumber, then seems fine for me